src/base/utils/string.cpp

```diff
@@ -126,12 +126,40 @@ namespace
 #endif
 
                 if (numL != numR)
-                    return (numL < numR);
+                    return (numL < numR) ? -1 : 1;
 
                 // Strings + digits do match and we haven't hit string end
                 // Do another round
             }
-            return false;
+            return 0;
+        }
+
+        static int naturalCompareCaseSensitive(const QString &left, const QString &right)
```

Comment:
I'm not sure if there is this rule: for each local function, you should create a function prototype